src/app.tsx

```diff
@@ -12,47 +14,104 @@ type Props = {
 // Type ViewState specifies the state (auth, bucket configuration, file upload display, etc)
 type State = {
     view: string;
-    s3Obj: MyS3Auth | {};
+    s3Obj: MyS3Auth | null;
 }
 
-// Driver for the UI
 // App will serve as the root node for the "tree" of different UIs. It will always render the "state" that is set by any sub function  
 export class App extends React.Component<Props, State>{
 
     // Create the App, set props and default the state of App to render auth
     constructor(props:Props) {
         super(props);
         this.setViewState = this.setViewState.bind(this);
-        this.setS3Obj = this.setS3Obj.bind(this)
+        this.setS3Obj = this.setS3Obj.bind(this);
         this.state = {
             view: 'auth',
-            s3Obj: {}
+            s3Obj: null
         };
+
+        // "Reconstruct" the App object if necessary
+        ViewStateStorage.getViewState(this.updateViewStatefromStorage);
+        UserMetaStorage.getUserS3Obj(this.updateS3ObjFromStorage);
     }
 
     // Function that will be passed as a prop to update the state of the UI
     setViewState(view:string) {
-        this.setState({view})
+        this.setState({view});
+        // Update state in chrome storage 
+        ViewStateStorage.putViewState(view);
     }
 
     // Function that will be passed as prop to globalize the S3 Object being made
-    setS3Obj(s3Obj:MyS3Auth){
-        this.setState({s3Obj})
+    setS3Obj(s3Obj:MyS3Auth | null){
+
+        this.setState({s3Obj});
+
+        // Change the user metadata only if the s3 object is not null
+        if (s3Obj){
```

Comment:
Should we say !== null for clarity?

Reply:
Actually that means something different 

see this https://stackoverflow.com/questions/4361585/how-to-check-if-a-variable-is-not-null

Reply:
imo I think its more concise